### ISSUE TITLE
Cannot change drawing behaviour by re-evaluating "draw" function in clojurescript

### ISSUE THREAD

**Issue by sp3ctum** [quil/quil#157]
Hi, I noticed this project a couple of weeks ago and finally gave it a try. I'm absolutely thrilled but also absolutely new and, unfortunately stuck as well.

I have a canvas that I can draw stuff to, but when I change my `draw` function I don't see the change in my browser window which I have open. I use figwheel to compile and reload the javascript on the page after each save. Evaluating the forms by hand does not change the result either.

What could I be doing wrong?

Here is my code:

```lisp
(ns codejam-practice-solutions.core
  (:require [quil.core :as q :include-macros true]))

(enable-console-print!)

(defn draw []
  (print "draw called at" (js/Date.)))

(defn setup []
  (q/frame-rate 1)
  (print "setup called at" (js/Date.)))

(q/defsketch my-sketch
  :host "canvas"
  :draw draw
  :setup setup
  :size [300 300])
```

**Comment by sp3ctum** [comment]
When I run my app I see a black canvas element appear, and the message by `draw` outputted to the console every second. The setup function is called in the very beginning, but not after a figwheel reload / manual C-c C-k reload (I use emacs).

**Comment by sp3ctum** [comment]
Also, after changing `draw`'s log message and re-evaluating it, I can see the changed result with a form like this:

```lisp
(q/with-sketch (q/get-sketch-by-id "canvas")
  (draw))
```

But it is displayed only once. While the `draw` function is run the original `draw` persists in logging the original message to the console.

Also I forgot to mention I use `[quil "2.2.5"]`

**Comment by nbeloglazov** [comment]
Hi Mika
Currently clojurescript version of quil doesn't have built-in support for live reloading. But I think you can do it pretty easily yourself. Try wrapping `draw` into anonymous function:
```clojure
(q/defsketch my-sketch
  :host "canvas"
  :draw #(draw)
  :setup setup
  :size [300 300])
```
This quil will always be calling `codejam-practice-solutions.core/draw` and reloading should work.

**Comment by nbeloglazov** [comment]
Please say if it helps. I don't use figweel or any other tool for live reloading of clojurescript so I can't check. If it works - we probably should include it into quil itself.

**Comment by sp3ctum** [comment]
Ah, good idea. I already tried #'var-quoting my draw function but it
changed nothing.

In addition to that, I found this example project that claims this should
be possible:
https://github.com/ClojureTO/hello-quil

Looks like that project does basically the same thing as I did but has its
defsketch and draw forms in different files. I'll investigate and report
back.

2015-05-29 7:24 GMT+03:00 Nikita Beloglazov <notifications@github.com>:

**Comment by sp3ctum** [comment]
Yep, sure enough, your method works perfectly - wrapping the function calls in an anonymous function allows them to be reloaded like everything else with figwheel. Thank you!

**Comment by nbeloglazov** [comment]
Great. I'll add it to quil itself.

**Comment by nbeloglazov** [comment]
Mika, could you try version 2.2.6-SNAPSHOT? I added live reloading to clojurescript version and would be glad if you tested to make sure it works. 

Thanks!

**Comment by sp3ctum** [comment]
Hi again, I finally got a chance to try this out. It seems to work in 2.2.6-SNAPSHOT right out of the box. :)

**Comment by nbeloglazov** [comment]
Cool, thanks for confirmation. I released 2.2.6 which contains the changes.
